src/static/js/webmentions.js

```diff
@@ -0,0 +1,331 @@
+// Code related to parsing and showing webmentions
+
+// Gets the webmentions json data from current URL
+async function getWebmentions(targetURL) {
+  const apiURL = `https://webmention.io/api/mentions.jf2?perPage=500&target=${targetURL}`;
+  let mentions = []
+  try {
+    const response = await window.fetch(apiURL);
+    if (response.status >= 200 && response.status < 300) {
+      const json = await response.json();
+      mentions = json.children;
+    } else {
+      console.error("Could not parse response", response.statusText);
+    }
+  } catch(error) {
+    console.error("Request failed", error);
+  }
+  return mentions;
+}
+
+// Format published date into human readable form
+function formatDate(dateString){
+  const options = { year: "numeric", month: "long", day: "numeric" }
+  return new Date(dateString).toLocaleDateString(undefined, options)
+}
+
+// Parse webmentions for individual category
+function parseMentions(webmentions, mentionType) {
+  let filteredMentions = []
+  webmentions.forEach(function(mention){
+    if (mention["wm-property"] == mentionType) {
+      filteredMentions.push(mention)
+    }
+  });
+  return filteredMentions;
+}
+
+// Renders webmention likes
+function renderLikes(likes) {
+  // Add the likes count to the likes-tab
+  document.querySelector('#likes-count').textContent = likes.length;
+  if (likes.length === 1) {
+    document.querySelectorAll('.like-singular').forEach(el => {
+      el.removeAttribute('data-translation');
+    });
+  } else {
+    document.querySelectorAll('.like-plural').forEach(el => {
+      el.removeAttribute('data-translation');
+    });
+  }
+
+  const likePanel = document.querySelector("#likes-panel");
+  const likeAriaText = likePanel.getAttribute("data-like-text");
+  const webmentionLikesList = document.createElement("ul");
+  webmentionLikesList.setAttribute("class", "webmention-likes");
+  likes.forEach(function(like) {
+    const likeHtml = `
+      <li class="webmention-likes-item">
+        <a
+          class="webmention-author"
+          href="${like["url"]}"
+          title="${like["author"]["name"]}"
+          aria-label="${like["author"]["name"]} ${likeAriaText}"
+        >
+          <img
+            class="webmention-author-avatar"
+            src="${like["author"]["photo"]}"
+            loading="lazy" width="60" height="60"/>
+        </a>
+      </li>
+    `;
+    const parser = new DOMParser();
+	  const likeHtmlObject = parser.parseFromString(likeHtml, 'text/html');
```

Comment:
What parts of the web mention are important to render as raw HTML? Why not play it safe and render text contents only?

Reply:
I thought about that, and there is a text field in the WebMentions data set. The only thing that renders is Links (including Twitter names). It's nice, but I'm not sure it's "have to add DomPurify to the site" nice.

HTML:

![image](https://user-images.githubusercontent.com/10931297/142954067-db0b8433-b6c3-412b-93f2-96e1c6fcf185.png)

Text:

![image](https://user-images.githubusercontent.com/10931297/142954230-17faf429-0d33-4439-b52e-3a287cc8f5d9.png)

Though currently the way the whole section is rendered is as a string, which is then parsed to HTML. So that would have to be rewritten but think we probably shoud.

If we are going that route we'd need to ensure w're using [`textcontent`](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) and not `innerHTML` or `innerText`.

Reply:
As mentioned in the description, I was also thinking about how to sanitize html or to use the text itself. In case we don't care about links, I think we can safely just show the text, given we anyways link to the source. In that case I can rewrite the entire renders using createElement, appendChild and textContent. I will do that.